Sound/Tidal/EspGrid.hs

```diff
@@ -0,0 +1,114 @@
+module Sound.Tidal.EspGrid where
+
+import Control.Concurrent.MVar
+import Control.Concurrent
+import Control.Monad (forever)
+import Control.Monad.Loops (iterateM_)
+import Sound.OSC.FD
+import Data.Time.Clock
+import Data.Time.Clock.POSIX
+import Data.Time.Calendar (fromGregorian)
+
+import Sound.Tidal.Tempo
+import Sound.Tidal.Time as T
+import Sound.Tidal.Stream
+import Sound.Tidal.Dirt
+import Sound.Tidal.Transition (transition)
+import Sound.Tidal.Pattern (silence)
+
+parseEspTempo :: [Datum] -> Maybe Tempo
+parseEspTempo d = do

```

Comment:
It's probably more idiomatic to say ```parseEspTempo [on', bpm', t1', t2', n'] = do```, and then referring to the args by name below instead of using ```(!!)```